### PR TITLE
CLOUD-475 Store only last 20 statuses of the cluster

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| self-healing | passed |
| big-data | passed |
| recreate | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| demand-backup | passed |
| limits | passed |
| operator-self-healing | passed |
| upgrade | failed |
| upgrade-consistency | passed |
| monitoring | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| recreate | passed |
| scaling | passed |
| big-data | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| limits | passed |
| one-pod | passed |
| scheduled-backup | passed |
| monitoring | passed |
| auto-tuning | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | failed |
| self-healing | passed |
| big-data | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| demand-backup | passed |
| limits | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing-advanced | failed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| self-healing-advanced | failed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| limits | passed |
| upgrade | passed |
| operator-self-healing | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| scheduled-backup | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| auto-tuning | failed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade | passed |
| limits | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| one-pod | passed |
| scheduled-backup | failed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | failed |
| storage | failed |
| scaling-proxysql | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring | passed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| big-data | passed |
| scaling-proxysql | passed |
| storage | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| upgrade | passed |
| limits | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-325-7ec03939

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| big-data | passed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |

File: pkg/controller/pxc/status.go [hunk 1]
```diff
@@ -18,15 +18,23 @@ import (
 )
 
 func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluster, reconcileErr error) (err error) {
+	maxStatusesQuantity := 20
```
[the remark this answers is not in the document]
Comment (cap1984) [reply pkg/controller/pxc/status.go]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| self-healing | passed |
| recreate | failed |
| init-deploy | passed |
| scaling | passed |
| storage | failed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| demand-backup | failed |
| limits | failed |
| upgrade | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| one-pod | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | failed |
| recreate | failed |
| limits | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup | failed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| recreate | passed |
| big-data | passed |
| limits | passed |
| demand-backup | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-325-a7ef7431

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| big-data | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scaling | passed |
| proxysql-sidecar-res-limits | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |

**Pull request merged**